botorch/generation/gen.py

```diff
@@ -69,11 +69,14 @@ def gen_candidates_scipy(
         equality constraints: A list of tuples (indices, coefficients, rhs),
             with each tuple encoding an inequality constraint of the form
             `\sum_i (X[indices[i]] * coefficients[i]) = rhs`.
-        nonlinear_inequality_constraints: A list of callables with that represent
-            non-linear inequality constraints of the form `callable(x) >= 0`. Each
+        nonlinear_inequality_constraints: A list of tuples, one tuple per
+            constraint, first element of the tuple is a callable that represents
+            a non-linear inequality constraint of the form `callable(x) >= 0`. Each
             callable is expected to take a `(num_restarts) x q x d`-dim tensor as
             an input and return a `(num_restarts) x q`-dim tensor with the
-            constraint values. The constraints will later be passed to SLSQP.
+            constraint values. The second element is a boolean indicating it is
+            an intrapoint or interpoint constraint. `True` for intrapoint,
+            `False` for interpoint. The constraints will later be passed to SLSQP.
```

Comment:
We should explain what the difference between interpoint and intrapoint constraint are, both here and in other places. We could refer to some other docstring if this would get to long as part of this one.

Reply:
Fully agree, still thinking where to do this in the best way ...

Reply:
I also updated the doc strings.